docs/source/notes/modules.rst

```diff
@@ -0,0 +1,433 @@
+Before we get to that, however, let's first examine how modules can be composed with one another.
+
+Modules as Building Blocks
+--------------------------
+
+Modules can contain other modules, making them useful building blocks for developing more complex functionality.
+The simplest way to do this is using the :class:`~torch.nn.Sequential` module. It allows us to chain together
+multiple modules:
+
+.. code-block:: python
+
+   net = nn.Sequential(
+     MyLinear(4, 3),
+     nn.ReLU(),
+     MyLinear(3, 1)
+   )
+
+   sample_input = torch.randn(4)
+   net(sample_input)
+   : tensor([-0.6749], grad_fn=<AddBackward0>)
+
+Note that :class:`~torch.nn.Sequential` automatically feeds the output of the first ``MyLinear`` module as input
+into the :class:`~torch.nn.ReLU`, and the output of that as input into the second ``MyLinear`` module. As
+shown, it is limited to linear chaining of modules.
+
+In general, it is recommended to define a custom module for anything beyond the simplest use cases, as this gives
+full flexibility on how submodules are used for a module's computation.
+
+For example, here's a simple neural network implemented as a custom module:
+
+.. code-block:: python
+
+   import torch.nn.functional as F
+
+   class Net(nn.Module):
+     def __init__(self):
+       super().__init__()
+       self.l0 = MyLinear(4, 3)
+       self.l1 = MyLinear(3, 1)
+     def forward(self, x):
+       x = self.l0(x)
+       x = F.relu(x)
+       x = self.l1(x)
+       return x
+
+This module is composed of two "children" or "submodules" (\ ``l0`` and ``l1``\ ) that define the layers of
+the neural network and are utilized for computation within the module's ``forward()`` method. Immediate
+children of a module can be iterated through via a call to :func:`~torch.nn.Module.children` or
+:func:`~torch.nn.Module.named_children`:
+
+.. code-block:: python
+
+   net = Net()
+   for child in net.named_children():
+     print(child)
+   : ('l0', MyLinear())
+   ('l1', MyLinear())
+
+To go deeper than just the immediate children, :func:`~torch.nn.Module.modules` and
+:func:`~torch.nn.Module.named_modules` *recursively* iterate through a module and its child modules:
+
+.. code-block:: python
+
+   class BigNet(nn.Module):
+     def __init__(self):
+       super().__init__()
+       self.l1 = MyLinear(5, 4)
+       self.net = Net()
+     def forward(self, x):
+       return self.net(self.l1(x))
+
+   big_net = BigNet()
+   for module in big_net.named_modules():
+     print(module)
+   : ('', BigNet(
+     (l1): MyLinear()
+     (net): Net(
+       (l0): MyLinear()
+       (l1): MyLinear()
+     )
+   ))
+   ('l1', MyLinear())
+   ('net', Net(
+     (l0): MyLinear()
+     (l1): MyLinear()
+   ))
+   ('net.l0', MyLinear())
+   ('net.l1', MyLinear())
+
+Sometimes, it's necessary for a module to dynamically define submodules.
+The :class:`~torch.nn.ModuleList` and :class:`~torch.nn.ModuleDict` modules are useful here; they
+register submodules from a list or dict:
```

Comment:
again, "registering" has not been explained.

Reply:
Fixed by explaining it above.